node/agent.go

```diff
@@ -394,7 +394,7 @@ func (a *HostAgent) Start(shutdown <-chan interface{}) {
 		// 1) has a connection
 		// 2) its node is registered
 		// 3) receives signal to shutdown or breaks
-		hsListener := zkservice.NewHostStateListener(a, a.hostID)
+		hsListener := zkservice.NewHostStateListener(a, a.hostID, shutdown)
```

Comment:
This needs to be outside of the for loop to keep state between any zzk failures.